Content.Server/Animals/Systems/EggLayerSystem.cs

```diff
@@ -60,11 +66,12 @@ public override void Update(float frameTime)
     private void OnMapInit(EntityUid uid, EggLayerComponent component, MapInitEvent args)
     {
         _actions.AddAction(uid, ref component.Action, component.EggLayAction);
-        component.CurrentEggLayCooldown = _random.NextFloat(component.EggLayCooldownMin, component.EggLayCooldownMax);
+        component.NextGrowth = TimeSpan.FromSeconds(_random.NextFloat(component.EggLayCooldownMin, component.EggLayCooldownMax));
```

Comment:
MapInit runs when the entity spawns, so this should use curtime + the random float, otherwise, after a few minutes have passed in the round, they're going to lay eggs immediately on spawning.